express-notes-api/index.js

```diff
@@ -0,0 +1,158 @@
+
+var express = require('express');
+var app = express();
+var filedata = require('./data.json');
+var fs = require('fs');
+const { restart } = require('nodemon');
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Express server listening on port 3000');
+});
+
+/*** errors */
+
+function error400PositiveInteger (res) {
+  res.status(400);
+  res.json({ "error": "id must be a positive integer" });
+}
+
+function error404CantFindID(res, id) {
+  res.status(404);
+  res.json({ "error": `cannot find note with id ${id}` });
+}
+
+function error404ContentRequiredField (res) {
+  res.status(400);
+  res.json({ "error": "content is a required field" });
+}
+
+function unexpectedError500 (res) {
+  res.status(500);
+  res.json({ "error": "An unexpected error occurred." });
+}
+
+
+/*** Clients can GET a list of notes */
+
+app.get('/api/notes', function (req, res) {
+  var arrayNotes = [];
+  for (const value in filedata.notes){
+    arrayNotes.push(filedata.notes[value]);
+  }
+  res.json(arrayNotes);
+});
+
+/** Clients can GET a single note by id */
+
+app.get('/api/notes/:id', function (req, res) {
+  var isThereANote = false;
+  var id = parseInt(req.params.id);
+
+  if (Number.isNaN(id)) {
+    error400PositiveInteger(res);
```

Comment:
These routes should respond with a `400` error for _any_ `id` which isn't a positive integer:

---

![image](https://user-images.githubusercontent.com/16143771/143334475-87e5d900-fc80-481d-9216-aca6e8a3950c.png)